eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -184,6 +185,7 @@ jobs:
           jdkArchitectureOption: 'x64'
           publishJUnitResults: false
           goals: 'install'
+          condition: ne(variables['Build.Reason'], 'PullRequest')
```

Comment:
I think we should continue running the SpotBugs Aggregate step, I've generally seen it catch things that the SpotBugs and Checkstyle step doesn't (usually the same issues that the site:site site:stage step catches).

Reply:
The reason for this is because ` 'Run SpotBugs, CheckStyle, and JavaDoc'` step doesn't include non-shipping-modules. If we add `-Dinclude-non-shipping-modules` to `options` in that step, then this step won't be required.

This step uses `eng/spotbugs-aggregate-reports/pom.xml` which is not included if the above flag is not set.

Reply:
Does this force us to also have to run 'install' on the earlier step?

Reply:
Before installing the clients the shipping modules are installed, should be a safe change to make.

Reply:
Analyze runs these steps in the following order:

1. Install Reporting Tools
2. Install SDKs
3. Checkstyle, SpotBugs, and Javadocs (changed in this PR)

It should be safe to add the `-Dinclude-non-shipping-modules` as an option in this step and be able to remove the `SpotBugs Aggregate Analysis` step that runs further down the line. If we opt to keep both and them being their own unique steps I would highly recommend moving the SptBugs Aggregate step before the `Generate Maven Report` step as both generally will fail with the same issues but the SpotBugs step is more verbose on the problem and will fail faster leading to better turnaround times.